File: Old AOTD/src/main/java/com/DavidM1A2/AfraidOfTheDark/common/entities/SplinterDrone/AnimationHandlerSplinterDroneProjectile.java
package com.DavidM1A2.AfraidOfTheDark.common.entities.SplinterDrone;

import java.util.HashMap;

import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.IMCAnimatedEntity;
import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.animation.AnimationHandler;
import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.animation.Channel;

public class AnimationHandlerSplinterDroneProjectile extends AnimationHandler
{
	/** Map with all the animations. */
	public static HashMap<String, Channel> animChannels = new HashMap<String, Channel>();

	static
	{
		animChannels.put("Sping", new ChannelSping("Sping", 100.0F, 100, Channel.LINEAR));
	}

	public AnimationHandlerSplinterDroneProjectile(IMCAnimatedEntity entity)
	{
		super(entity);
	}

	@Override
	public void activateAnimation(String name, float startingFrame)
	{
		super.activateAnimation(animChannels, name, startingFrame);
	}

	@Override
	public void stopAnimation(String name)
	{
		super.stopAnimation(animChannels, name);
	}
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/entity/splinterDrone/animation/AnimationHandlerSplinterDroneProjectile.kt
package com.davidm1a2.afraidofthedark.common.entity.splinterDrone.animation

import com.davidm1a2.afraidofthedark.common.entity.mcAnimatorLib.animation.AnimationHandler
import com.davidm1a2.afraidofthedark.common.entity.mcAnimatorLib.animation.ChannelMode

/**
 * Animation handler class for the splinter drone projectile
 */
class AnimationHandlerSplinterDroneProjectile : AnimationHandler(
    setOf(
        ChannelSping("Sping", 100.0f, 100, ChannelMode.LINEAR)
    )
)